qiita_db/processing_job.py

```diff
@@ -86,12 +86,18 @@ def __init__(self):
     def _element_extract(self, snippet, list_of_elements,
                          list_of_optional_elements):
         results = {}
+        missing_elements = []
+
         for element in list_of_elements:
             value = search('<%s>(.*?)</%s>' % (element, element), snippet)
             if value:
                 results[element] = value.group(1)
             else:
-                raise AssertionError("%s element is missing!" % element)
+                missing_elements.append(element)
+
+        if missing_elements:
+            raise AssertionError("The following elements were not found: %s"
+                                 % str(missing_elements))
```

Comment:
Can you format the list as: `', '.join(missing_elements)` and remove the `str` cast.

Reply:
You got it.